pkg/controllers/updaterun/controller.go

```diff
@@ -31,9 +32,26 @@ import (
 	"go.goms.io/fleet/pkg/utils"
 	"go.goms.io/fleet/pkg/utils/condition"
 	"go.goms.io/fleet/pkg/utils/controller"
+	"go.goms.io/fleet/pkg/utils/controller/metrics"
 	"go.goms.io/fleet/pkg/utils/informer"
 )
 
+// updateRunMetricsStatus is the status string for update run metrics.
+type updateRunMetricsStatus string
+
+const (
+	// updateRunMetricsStatusInitializing represents the update run is initializing in the metrics.
+	updateRunMetricsStatusInitializing updateRunMetricsStatus = "initializing"
+	// updateRunMetricsStatusProgressing represents the update run is in progress in the metrics.
+	updateRunMetricsStatusProgressing updateRunMetricsStatus = "progressing"
```

Comment:
As per offline discussion, can we either have progressing/stuck metric at any given point where we delete,
- progressing -> stuck, only emit stuck metric, delete progressing metric
- stuck -> progressing, only emit progressing, delete stuck metric

Instead of both metric lines at the same time, unless there is a need to having them both at the same time

Reply:
updated to only emit either one.

Reply:
if we emit the stuck & processing in different time, better to keep both instead of deleting. That's another data point.